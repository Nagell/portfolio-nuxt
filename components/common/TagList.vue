<template>
    <div
        v-if="hasTags"
        class="flex flex-wrap gap-2"
    >
        <Badge
            v-for="(tag, key) in tags"
            :key="key"
            variant="muted"
        >
            {{ tag }}
        </Badge>
    </div>
</template>

<script lang="ts" setup>
    import type { Json } from '~/types/database.types'

    interface Props {
        tags: Json
    }

    const props = defineProps<Props>()

    /** Check if there are any tags */
    const hasTags = computed(() => Array.isArray(props.tags) && props.tags.length > 0)
</script>
