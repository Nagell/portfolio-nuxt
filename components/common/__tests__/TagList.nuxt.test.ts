import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import TagList from '../TagList.vue'

describe('TagList Component', async () => {
    it('renders correctly when tags are provided', async () => {
        const tags = [ 'Vue', 'Nuxt', 'TypeScript' ]
        const wrapper = await mountSuspended(TagList, {
            props: { tags }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders no content when tags are not provided', async () => {
        const wrapper = await mountSuspended(TagList, {
            props: { tags: [] }
        })
        const div = wrapper.find('div')

        expect(div.exists()).toBe(false)
    })

    it('renders all tags correctly', async () => {
        const tags = [ 'Vue', 'Nuxt', 'TypeScript' ]
        const wrapper = await mountSuspended(TagList, {
            props: { tags }
        })
        const badges = wrapper.findAllComponents({ name: 'Badge' })

        expect(badges.length).toBe(tags.length)
        tags.forEach((tag, index) => {
            expect(badges[index].text()).toBe(tag)
        })
    })
})
