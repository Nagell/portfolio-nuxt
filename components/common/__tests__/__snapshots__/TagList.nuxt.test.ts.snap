// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagList > renders correctly when tags are provided 1`] = `
"<div class="flex flex-wrap gap-2">
  <div class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground">Vue</div>
  <div class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground">Nuxt</div>
  <div class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground">TypeScript</div>
</div>"
`;
